compiler/hie-core/src/Development/IDE/Core/Rules.hs

```diff
@@ -131,8 +132,9 @@ writeIfacesAndHie ::
 writeIfacesAndHie ifDir files =
     runMaybeT $ do
         tcms <- usesE TypeCheck files
-        session <- lift $ useNoFile_ GhcSession
-        liftIO $ concat <$> mapM (writeTcm session) tcms
+        fmap concat $ forM (zip files tcms) $ \(file, tcm) -> do
```

Comment:
This looks somewhat fishy. The way we call `writeIfacesAndHie` it is only called in cases where `files` corresponds to the transitive dependency closure of a file. If those have different `HscEnv`s something is off afaict so it would be nice to check that and throw an error if it breaks.

Reply:
If things have got this far then creating the interfaces with different HscEnv's worked, so it's reasonable to continue successfully. I think the check for compatibility should come in type checking, since that's where we combine things from different HscEnv's into a single type checked module, which is the thing that is more likely to fail. (However, talking to some GHC devs, they thought it might actually work - but it's something that needs testing)